actors/miner/src/lib.rs

```diff
@@ -4556,13 +4563,13 @@ fn termination_penalty(
         let sector_power = qa_power_for_sector(sector_size, sector);
         let fee = pledge_penalty_for_termination(
             &sector.expected_day_reward,
-            current_epoch - sector.activation,
+            current_epoch - sector.power_base_epoch,
             &sector.expected_storage_pledge,
             network_qa_power_estimate,
             &sector_power,
             reward_estimate,
             &sector.replaced_day_reward,
```

Comment:
So I might be missing this but I don't see the update to replaced_day_reward happening in extension of simple qap sectors.  As it stands it looks like replaced_day_reward stays at initial value of 0 for expiration extension and so termination fee undercounts all of previous sector lifetime.

The other thing here is that now that replaced day reward can be updated many times we need to figure out how to track many changes which you talk about in #914 but I am missing here too.  Setting replaced_day_reward to max of replaced_day_reward and day_reward is my preference.  This needs to happen both in expiration extension and in replica update.

Reply:
[This looks done now ](https://github.com/filecoin-project/builtin-actors/pull/1229/files#diff-fbcb2ec1a9d82b18f146c728cafd643df0e7ae47a04d84be7644913fe89236e5R3851)